File: python/p2.py
#Q: Find the sum of all the even-valued terms in the Fibonacci sequence which do not exceed four million.
#A: 4613732

def fib(n):
    result = []
    a, b = 0, 1
    while b < n:
        result.append(b)
        a, b = b, a+b
    return result

print sum([i for i in fib(4000001) if i%2==0])

File: clj/p2.clj
;;Q: Find the sum of all the even-valued terms in the Fibonacci sequence which do not exceed four million.
;;A: 4613732

(loop [a 0 b 1 sum 0]
  (if (>= b 4000000)
    sum
    (recur b (+ a b) (if (zero? (rem b 2))
		       (+ sum b)
		       sum))))